src/main/org/ovirt/jenkins/plugin/OvirtCloud.java

```diff
@@ -0,0 +1,81 @@
+package main.org.ovirt.jenkins.plugin;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import hudson.model.Label;
+import hudson.slaves.Cloud;
+import hudson.slaves.NodeProvisioner;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+
+public class OvirtCloud extends Cloud {
+
+    private static final Logger LOGGER = Logger.getLogger(OvirtCloud.class.getName());
+
+    private Label label;
+
+    public Label getLabel() { return this.label; }
+
+    @DataBoundConstructor
+    public OvirtCloud(String name) {
+        super(name);
+    }
+
+    public boolean canProvision(Label label) { return this.label == label; }
+
+    public Collection<NodeProvisioner.PlannedNode> provision(final Label label, final int excessWorkload) {
+
+        LOGGER.log(Level.INFO, "Provisioning {0} VMs", excessWorkload);
+
+        List<NodeProvisioner.PlannedNode> provisionedList = new ArrayList<>();
+
+        for (int i = 0; i < excessWorkload; ++i) {
+            try {
+                oVirtProvision(label);
+            } catch (Exception e) {
+                // TODO: Do something here?
+            }
+        }
+
+
+        return provisionedList;
+    }
+
+
+    /**
+     * Provision an oVirt VM to connect back to Jenkins
+     *
+     * @param label
+     * @return true if successfully provisioned oVirt VM
+     * @throws Exception if something went wrong
+     */
+    private boolean oVirtProvision(Label label) throws Exception {
```

Comment:
I guess this is our private implementation, so we don't need the oVirt prefix, just "provision" or "provisionVM"